subworkflows/primaryanalysis.nf

```diff
@@ -120,5 +122,20 @@ workflow PRIMARY_ANALYSIS {
 
     ch_icount_peaks = GET_CROSSLINKS.out.crosslinkBed.combine(ICOUNT_SIGXLS.out.sigxls, by: 0)
     ICOUNT_PEAKS ( ch_icount_peaks )
-    
+
+    // If genome is compressed, uncompress it
+    if (params.fasta.matches(".*gz")) {
+        ch_fasta = genome.map{ folder -> file(folder + "/DNA_GUNZIP/*.fa")}
+    } else {
+        ch_fasta = file(params.fasta)
+    }
```

Comment:
I'm not totally happy about having this `if` block here, but can't think of a cleaner way of doing it.

Reply:
I think I'd prefer not to have primaryanalysis taking a fasta param. The genome folder should have the preparegenome input files in there too, so I was thinking of having it do 'if there's a DNA_GUNZIP folder, use the fasta file in there, otherwise get the original fasta input from the folder' - do you think that would cause any issues?

Reply:
Yep, that seems good. In which case, the `test` profiles can be removed from the testing runs.